system/handlers/dbmigrations/emailTemplateActivityClickStatHash.cfc

```diff
@@ -0,0 +1,122 @@
+component {
+
+	private void function run() {
+		var presideObjectService = getController().getWirebox().getInstance( "presideObjectService" );
+		var dbInfoService        = getController().getWirebox().getInstance( "dbInfoService" );
+		var sqlRunner            = getController().getWirebox().getInstance( "sqlRunner" );
+		var objectName           = "email_template_send_log_activity";
+		var dbAdapter            = presideObjectService.getDbAdapterForObject( objectName );
+		var dsn                  = presideObjectService.getObjectAttribute( objectName, "dsn" );
+		var tableName            = presideObjectService.getObjectAttribute( objectName, "tablename" );
+		var dbInfo               = dbInfoService.getDatabaseVersion( dsn );
+		var escapedTableName     = dbAdapter.escapeEntity( tableName );
+		var updateHashCols       = [];
+		var addTempColSqlArray   = [];
+		var dropTempColSqlArray  = [];
+		var addTempIndexSql      = "";
+		var dropTempIndexSql     = "";
+
+		for ( var rawCol in [ "link", "link_title", "link_body" ] ) {
+			var escapedRawCol      = dbAdapter.escapeEntity( rawCol );
+			var escapedHashCol     = dbAdapter.escapeEntity( "#rawCol#_hash" );
+			var escapedHashTempCol = dbAdapter.escapeEntity( "#rawCol#_hash_temp" );
+
+			var addTempColSql  = "";
+			var dropTempColSql = "";
+
+			switch ( dbInfo.database_productName ) {
```

Comment:
There is an issue here in that this requires the database user to have DDL privileges. For production instances, it is advised that the db user for the application only has data privileges and is not able to update the schema (and this is the case at Pixl8, for example, which means this migration will fail).

Did you consider running something like the following instead?:

```sql
update #tablename#
set #hashCol# = md5(#sourceCol#)
where #hashCol# is null and #sourceCol# is not null
```

Reply:
Yup, that make sense Dom. 
Have updated the dbmigration script. 

Also updated filter in the getLinkClickStats which was missing the null check.